ConnectMessage/index.js

```diff
@@ -1,175 +1,168 @@
-        const client = ns.createQueueClient(queueName)
-            , sender = client.createSender()
-            , message = {body: {test: test, xml: rawXML}}
-            ;
-        await sender.send(message);
-        await client.close();
-    }
-    catch (e) {
-        error = e
-    }
-    finally {
-        if (ns) {await ns.close()}
-    }
-    return error
-}
+'use strict';
+
+const crypto = require('crypto')
+    , { ServiceBusClient } = require("@azure/service-bus")  // See https://github.com/Azure/azure-sdk-for-js/tree/master/sdk/servicebus/service-bus/
+    ;
+
+const debug = true;
+
+const sleep = (milliseconds) => {
+    return new Promise(resolve => setTimeout(resolve, milliseconds))
+}
+
+module.exports.connectNotificationMessage = async function (context, req) {
+  context.log('JavaScript HTTP trigger function processed a request.');
+
+  const debugLog = msg => {if (debug) {context.log(msg)}};
+  const invocationId = context.executionContext && context.executionContext.invocationId;
+
+  function checkBasicAuth() {
+      const name = process.env['BASIC_AUTH_NAME']
+          , pw = process.env['BASIC_AUTH_PW']
+          , authRaw0 = (req.headers && req.headers.authorization) || ''
+          , authRaw = authRaw0.split(' ')[1]  || ''
+          , authString = Buffer.from(authRaw, 'base64').toString()
+          , authArray = authString.split(':')
+          , authenticated = name == authArray[0] && pw == authArray[1]
+          ;
+      return authenticated
+  }
+  
+  debugLog(`Started!. Invocation ID: ${invocationId}`);
+  // Check Basic Authentication 
+  if (checkBasicAuth()) {
+      debugLog("Authenticated!")
+  } else {
+      const headers = {'WWW-Authenticate': 'Basic realm="Connect Listener", charset="UTF-8"'};
+      context.res = {status: 401, body: "Unauthorized!", headers: headers};
+      return // EARLY return  
+  }
+
+  // Check HMAC and enqueue. Allow for test messages
+  const test = req.query.test ? req.query.test : false
+      , rawXML = req.body
+      , hmac1 = process.env['HMAC_1']
+      , hmacConfigured = hmac1;
+
+  let payload = rawXML;
+
+  if (process.env['JSON']) {
+    payload = JSON.stringify(rawXMl);
+  }
+
+  let hmacPassed;
+  if (!test && hmacConfigured) {
+      // Not a test:
+      // Step 1. Check the HMAC
+      // get the headers
+      const authDigest = req.headers['x-authorization-digest']
+          , hmacSig1 = req.headers['x-docusign-signature-1']
+          ;
+      debugLog(`hmacSig1: ${hmacSig1}`)
+      hmacPassed = checkHmac(hmac1, payload, authDigest, hmacSig1)
+      if (!hmacPassed) {
+          context.log.error(`HMAC did not pass!! HMAC Sig 1: ${hmacSig1}`);
+          context.res = {status: 401, body: "Bad HMAC: unauthorized!"};
+          return // EARLY return      
+      }
+      if (hmacPassed){
+          debugLog('HMAC passed!')
+      }
+  } else {
+      // hmac is not configured or a test message. HMAC is not checked for tests.
+      hmacPassed = true
```

Comment:
not using ; because it's only one statement is not good practice as if you need to add another statement, you'll have to add it.

Reply:
Thanks, @InbarGazit. This is existing code, I will fix it and i see it happen in multiple other areas. I fixed the indentation to be consistent with the other two samples - that's why the diff shows this as a big change.